IdentityCore/src/requests/MSIDSilentTokenRequest.m

```diff
@@ -235,129 +216,148 @@ - (void)executeRequestImpl:(MSIDRequestCompletionBlock)completionBlock
-                         {
-                             [self tryAppRefreshToken:appRefreshToken completionBlock:completionBlock];
-                             return;
-                         }
-
-                         NSError *interactionError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInteractionRequired, @"User interaction is required", error.msidOauthError, error.msidSubError, error, self.requestParameters.correlationId, nil, YES);
-                         completionBlock(nil, interactionError);
-                     }
-                     else
-                     {
-                         completionBlock(nil, error);
-                     }
-                 }
-                 else
-                 {
-                     completionBlock(result, nil);
-                 }
-             }];
+    BOOL isAppRefreshToken = tokenType == MSIDAppRefreshTokenType;
+    MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, self.requestParameters, @"Trying to acquire access token using %@ for clientId %@, authority %@, account %@", (!isAppRefreshToken ? @"App Refresh Token" : @"Family Refresh Token"), self.requestParameters.authority, self.requestParameters.clientId, self.requestParameters.accountIdentifier.maskedHomeAccountId);
+    
+    // When using ART or FRT, it will go through the same method below, and handle differently within the completion block
+    [self redeemAccessTokenWith:refreshToken
+                completionBlock:^(MSIDTokenResult * _Nullable result, NSError * _Nullable error) {
+        if (error)
+        {
+            if ([self isErrorRecoverableByUserInteraction:error])
+            {
+                if (isAppRefreshToken)
+                {
+                    // Handle error case when try with App refresh token
+                    MSIDBaseToken<MSIDRefreshableToken> *familyRefreshToken = [self handleErrorResponseForAppRefreshToken:refreshToken];
+                    if (familyRefreshToken)
+                    {
+                        [self tryRefreshToken:familyRefreshToken
+                                    tokenType:MSIDFamilyRefreshTokenType
+                              completionBlock:completionBlock];
+                        return;
+                    }
+                }
+                
+                // Handle error case when try with Family refresh token
+                [self handleErrorResponseForFamilyRefreshToken:error];
+                NSError *interactionError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInteractionRequired, @"User interaction is required", error.msidOauthError, error.msidSubError, error, self.requestParameters.correlationId, nil, YES);
+                completionBlock(nil, interactionError);
+            }
+            else
+            {
+                completionBlock(nil, error);
+            }
+        }
+        else
+        {
+            completionBlock(result, nil);
```

Comment:
nit: consider doing this check very early to avoid one extra level if indentation

Reply:
Could you specify which check it is? The comment landed on a completionBlock block :)

Reply:
You can do if (!error) (call completionBlock) as the very first thing on line 289. This would made code a bit more readable.